hydradx/model/amm/concentrated_liquidity_pool.py

```diff
@@ -0,0 +1,162 @@
+from math import sqrt as sqrt
+import math
+from .agents import Agent
+from .amm import AMM
+# from mpmath import mp, mpf
+# mp.dps = 50
+
+tick_increment = 1 + 1e-4
+def tick_to_price(tick: int):
+    return tick_increment ** tick
+
+def price_to_tick(price: float, tick_spacing: int = 1):
+    raw_tick = math.log(price) / math.log(tick_increment)
+    nearest_valid_tick = round(raw_tick / tick_spacing) * tick_spacing
+    return nearest_valid_tick
+
+import math
+
+class ConcentratedLiquidityState(AMM):
+    def __init__(
+            self,
+            assets: dict,
+            min_tick: int = None,
+            max_tick: int = None,
```

Comment:
Do we really want constructor to use ticks instead of prices? In actual implementation, the price to tick conversion would be done off-chain, so I can see why on-chain implementation would have this. But I don't know if it makes sense for us to implement it this way. We can still track ticks in the data structure, but maybe we should do the price -> tick conversion in the constructor?